redis/src/types.rs

```diff
@@ -2076,149 +2075,122 @@ from_vec_from_redis_value!(<T> Vec<T>);
 from_vec_from_redis_value!(<T> std::sync::Arc<[T]>);
 from_vec_from_redis_value!(<T> Box<[T]>; Vec::into_boxed_slice);
 
-impl<K: FromRedisValue + Eq + Hash, V: FromRedisValue, S: BuildHasher + Default> FromRedisValue
-    for std::collections::HashMap<K, V, S>
-{
-    fn from_redis_value(v: &Value) -> RedisResult<std::collections::HashMap<K, V, S>> {
-        let v = get_inner_value(v);
-        match *v {
-            Value::Nil => Ok(Default::default()),
-            _ => v
-                .as_map_iter()
-                .ok_or_else(|| {
-                    invalid_type_error_inner!(v, "Response type not hashmap compatible")
-                })?
-                .map(|(k, v)| Ok((from_redis_value(k)?, from_redis_value(v)?)))
-                .collect(),
-        }
-    }
-    fn from_owned_redis_value(v: Value) -> RedisResult<std::collections::HashMap<K, V, S>> {
-        let v = get_owned_inner_value(v);
-        match v {
-            Value::Nil => Ok(Default::default()),
-            _ => v
-                .into_map_iter()
-                .map_err(|v| invalid_type_error_inner!(v, "Response type not hashmap compatible"))?
-                .map(|(k, v)| Ok((from_owned_redis_value(k)?, from_owned_redis_value(v)?)))
-                .collect(),
-        }
-    }
-}
+macro_rules! impl_from_redis_value_for_map {
+    (for <$($TypeParam:ident),+> $MapType:ty, where ($($WhereClause:tt)+), err_msg: $err_msg:expr ) => {
```

Comment:
nit: IMO `err_msg` is unnecessary. Since there isn't a difference between the maps in their parsing process, this can just be "Response type not map compatible"